spec/support/shared_active_scaffold_page_behavior.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+# Very basic smoke test of ActiveScaffold components; click on each of
+# the actions and make sure it doesn't explode.
+shared_examples 'an ActiveScaffold page' do |actions, page_text|
+  describe 'the actions' do
+    before { visit(url) }
+
+    include_context 'log in super admin'
+
+    actions.each do |action|
+      describe action do
+        before do
+          first('td.actions').click_link action
+          page.driver.browser.switch_to.alert.accept if action == 'Delete'
+        end
+
+        it 'leaves us on the page' do
+          expect(page).to have_text(page_text)
+        end
+      end
+    end
+  end
+end
+
+shared_examples 'a Create New' do
+  context 'when a super admin is logged in', :js do
+    include_context 'log in super admin'
+
+    before do
+      visit(url)
+      click_link 'Create New'
+      sleep(0.5)
```

Comment:
Capybara is normally smart enough to wait for page elements, but for some reason the Active Scaffold pages don't play nicely with that feature. To get around that in our cucumber tests, we use a helper step:
```ruby
features/step_definitions/web_steps.rb

Then "I wait for ajax"  do
  Timeout.timeout(Capybara.default_wait_time) do
    loop do
      active = page.evaluate_script('jQuery.active')
      break if active == 0
    end
  end
end
```
I recommend moving the code to a helper for use by both RSpec and Cucumber:
```ruby
spec/support/ajax_helpers.rb

# In most cases, Capybara is smart enough to wait for page elements to be updated.
# This helper should only be used when absolutely necessary, such as for Active Scaffold pages.
def wait_for_ajax
  Timeout.timeout(Capybara.default_wait_time) do
    loop do
      active = page.evaluate_script('jQuery.active')
      break if active == 0
    end
  end
end
```

```ruby
features/support/env.rb

require_relative '../../spec/support/omniauth_helpers.rb'
require_relative '../../spec/support/ajax_helpers.rb'
```

```ruby
features/step_definitions/web_steps.rb

Then "I wait for ajax"  do
  wait_for_ajax
end
```
Then you can replace all the `sleep` calls with `wait_for_ajax`.

Reply:
Added.